Objects/unicodeobject.c

```diff
@@ -230,8 +230,19 @@ static inline PyObject* unicode_get_empty(void)
 // Return a strong reference to the empty string singleton.
 static inline PyObject* unicode_new_empty(void)
 {
-    PyObject *empty = unicode_get_empty();
-    Py_INCREF(empty);
+    PyObject *empty = NULL;
+    if (_Py_unicode_inited == 0) {
```

Comment:
This change removes my micro-optimization:

```
static int
unicode_create_empty_string_singleton(struct _Py_unicode_state *state)
{
    // Use size=1 rather than size=0, so PyUnicode_New(0, maxchar) can be
    // optimized to always use state->empty_string without having to check if
    // it is NULL or not.
```

I wrote PR #25147 which reorganizes Py_Initialize() to fix https://bugs.python.org/issue43687 without losing the optimization.